node_normalizer/server.py

```diff
@@ -105,13 +105,18 @@ async def get_conflations() -> ConflationList:
     summary='Get the equivalent identifiers and semantic types for the curie(s) entered.',
     description='Returns the equivalent identifiers and semantic types for the curie(s)'
 )
-async def get_normalized_node_handler(curie: List[str] = Query([], example=['MESH:D014867', 'NCIT:C34373']), conflate: bool = True):
+async def get_normalized_node_handler(curie: List[str] = Query([], example=['MESH:D014867', 'NCIT:C34373'], min_items=1), conflate: bool = True):
     """
     Get value(s) for key(s) using redis MGET
     """
     # no_conflate = request.args.get('dontconflate',['GeneProtein'])
     normalized_nodes = await get_normalized_nodes(app, curie, conflate)
 
+    # If curie contains at least one entry, then the only way normalized_nodes could be blank
+    # would be if an error occurred during processing.
+    if not normalized_nodes:
```

Comment:
if we are making the effort to raise an exception it may be helpful to add more context to the message.

Reply:
The main reason we need an exception here is to avoid returning `{}` with an HTTP 200 success code back to the client when an exception is raised within `get_normalized_nodes()` during processing -- something that shouldn't happen unless some unknown bug is triggered in this code. I'm not sure how to make the exception detail message more detailed, since the [try-catch block in `get_normalized_nodes()`](https://github.com/TranslatorSRI/NodeNormalization/blob/ebb8cf028a95af4e270cd5802a5742606bbd6acb/node_normalizer/normalizer.py#L466-L530) doesn't return any information on what triggered the exception to the calling method (it logs it and returns an empty dictionary instead). However, I could get rid of that try-catch block entirely: if I do that, then any exception raised in `get_normalized_nodes()` should cause FastAPI to log the exception and return an HTTP 500 Internal Server Error without the need for manually raising an HTTPException here. Would that be better?